modules/acquia_cms_search/acquia_cms_search.module

```diff
@@ -161,3 +162,80 @@ function acquia_cms_search_search_api_server_update(EntityInterface $entity) {
 function acquia_cms_search_form_acquia_cms_solr_search_form_alter(array &$form) {
   $form['#submit'][] = AcquiaSearchFacade::class . '::submitSettingsForm';
 }
+
+/**
+ * Implements hook_modules_installed().
+ */
+function acquia_cms_search_modules_installed(array $modules) {
+  $module_handler = Drupal::moduleHandler();
+  // Category facet is created if content type modules are installed.
+  if ($module_handler->moduleExists('acquia_cms_person') ||
+  $module_handler->moduleExists('acquia_cms_place') ||
+  $module_handler->moduleExists('acquia_cms_article') ||
+  $module_handler->moduleExists('acquia_cms_event') ||
+  $module_handler->moduleExists('acquia_cms_page')) {
+    // Load the category facet if it exists.
+    $category_facet = Facet::load('search_category');
+    // Create a new facet if the category facet doesn't exists.
+    if (empty($category_facet)) {
+      // Programmatically creating the facet.
+      $facet = Facet::create([
```

Comment:
@panshulK Let's not add create code here and create a separate class (so we can write PHPUnit tests) and here just add create facet code. (Probably create a class in src/Facade folder and could be name as FacetFacade).

Reply:
Done